powerstore/resource_storagecontainer.go

```diff
@@ -0,0 +1,187 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Storage Container",
+			"Could not create Storage Container, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Get Storage Container Details using ID retrieved above
+	storageContainerResponse, err1 := r.p.client.PStoreClient.GetStorageContainer(context.Background(), storageContainerCreateResponse.ID)
+	if err1 != nil {
+		resp.Diagnostics.AddError(
+			"Error getting Storage Container after creation",
+			"Could not get Storage Container, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := models.StorageContainer{}
+
+	updateStorageContainerState(&result, storageContainerResponse, &plan, "Create")
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	log.Printf("Done with Create")
+}
+
+// Read resource information
+func (r resourceStorageContainer) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+
+	var state models.StorageContainer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Storage Container details from API and then update what is in state from what the API returns
+	storageContainerID := state.ID.Value
+	storageContainerResponse, err := r.p.client.PStoreClient.GetStorageContainer(context.Background(), storageContainerID)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading storage container",
+			"Could not read storageContainerID with error "+storageContainerID+": "+err.Error(),
+		)
+		return
+	}
+
+	updateStorageContainerState(&state, storageContainerResponse, nil, "Read")
+
+	// Set state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Done with Read")
+}
+
+// Update resource
+func (r resourceStorageContainer) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+
+}
+
+// Delete resource
+func (r resourceStorageContainer) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
+	log.Printf("Started with Delete")
+
+	var state models.StorageContainer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get storage container ID from state
+	storageContainerID := state.ID.Value
+
+	// Delete Storage Container by calling API
+	_, err := r.p.client.PStoreClient.DeleteStorageContainer(context.Background(), storageContainerID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting storage container",
+			"Could not delete storageContainerID "+storageContainerID+": "+err.Error(),
+		)
+		return
+	}
+
+	// Remove resource from state
+	resp.State.RemoveResource(ctx)
```

Comment:
Please check if this statement is required

Reply:
removed